### PR TITLE
feat: add snakeToCamel util function

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### Description Of Changes
Applies https://sendbird.atlassian.net/browse/UIKIT-3974

The format of the UIKitConfiguration payload will consist of snake_case like [this](https://sendbird.atlassian.net/wiki/spaces/SDK/pages/2060618878/UIKit+Feature+configuration+RFC#JSON-Payload). So I made a utility function to convert it into camelCase, which is more natural in our application."

### CODE REVIEW

**Review by sravan-s — APPROVED** [review]

File: src/utils/__tests__/snakeToCamel.spec.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    expect(snakeToCamel(input)).toEqual(expectedOutput);
+  });
+
+  it('should handle empty objects and arrays', () => {
+    expect(snakeToCamel({})).toEqual({});
+    expect(snakeToCamel([])).toEqual([]);
+  });
+
+  it('should preserve non-object and non-array values', () => {
+    expect(snakeToCamel('test')).toBe('test');
```
Comment (sravan-s) [inline src/utils/__tests__/snakeToCamel.spec.ts:91]: suggestion: good to add test for `camel_case`,

File: src/utils/snakeToCamel.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+type CamelCase<T> = T extends object
```
Comment (sravan-s) [inline src/utils/snakeToCamel.ts:1]: 👍

File: src/utils/__tests__/snakeToCamel.spec.ts (hunk 1 above)
Comment (AhyoungRyu) [reply src/utils/__tests__/snakeToCamel.spec.ts:94]: Yeah I added it [here](https://github.com/sendbird/sendbird-uikit-react/pull/548/commits/53a68d8b91289f258aa0b88c884c75df5cd0d3a7#diff-f73dfe797e2645768392a250b19223f6cca5439911d3235815b4eb37e7f97cadR84).

File: src/utils/snakeToCamel.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    return obj as CamelCase<T>;
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map((item) => snakeToCamel(item)) as CamelCase<T>;
+  }
+
+  const result = {} as CamelCase<T>;
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
```
Comment (HoonBaek) [inline src/utils/snakeToCamel.ts:22]: Do we need to use the prototype function here? because It already checked the types above
(just curious)

**Review by HoonBaek — APPROVED** [review]
LGTM

File: src/utils/snakeToCamel.ts (hunk 3 above)
Comment (AhyoungRyu) [reply src/utils/snakeToCamel.ts:22]: @HoonBaek Where's the above?

Comment (HoonBaek) [reply src/utils/snakeToCamel.ts:22]: on the line numbers 12 and 16
Isn't already checked?

Comment (AhyoungRyu) [reply src/utils/snakeToCamel.ts]: Ahh this line is for checking if the object has the key not checking if its type is Object or Array. 

Also, this line might look unnecessary cause the `key` is from just its above like in for-in loop. 
You can find the detail of behind reason is described in [this Stackoverflow question](https://stackoverflow.com/questions/1963102/what-does-the-jslint-error-body-of-a-for-in-should-be-wrapped-in-an-if-statemen). 

But anyways, I also didn't like this pattern either. So I dropped the for-in + if pattern and replaced it \w `Object.keys.forEach(key => )`. 992def4bce33a54292532cc55703275e2c85a02a 

Thanks for mentioning it Hoon :)

**Pull request merged**